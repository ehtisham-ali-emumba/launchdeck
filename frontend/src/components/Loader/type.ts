export type LoaderProps = {
  marginTop?: string;
  paddingTop?: string;
};
