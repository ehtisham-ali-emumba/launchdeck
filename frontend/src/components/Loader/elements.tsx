import { Spin } from "antd";
import styled from "styled-components";

import { colors } from "../../constants";

export const CenteredWrapper = styled.div<{
  marginTop?: string;
  paddingTop?: string;
}>`
  display: flex;
  justify-content: center;
  align-items: flex-start;
  width: 100%;
  padding-top: ${props => props.paddingTop || "30px"};
  margin-top: ${props => props.marginTop || "0px"};
`;

export const OrangeSpin = styled(Spin)`
  .ant-spin-dot-item {
    background-color: ${colors.primary};
  }
`;
