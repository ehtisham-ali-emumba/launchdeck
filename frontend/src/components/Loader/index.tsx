import React from "react";

import { CenteredWrapper, OrangeSpin } from "./elements";
import type { LoaderProps } from "./type";

export const Loader: React.FC<LoaderProps> = ({ marginTop, paddingTop }) => (
  <CenteredWrapper marginTop={marginTop} paddingTop={paddingTop}>
    <OrangeSpin size="large" />
  </CenteredWrapper>
);
