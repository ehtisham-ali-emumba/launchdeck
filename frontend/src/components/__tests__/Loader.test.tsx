import { render, screen } from "@testing-library/react";

import { Loader } from "../Loader";

describe("Loader", () => {
  it("renders the spinner inside a centered wrapper", () => {
    render(<Loader marginTop={"10px"} paddingTop={"20px"} />);
    const spinner = screen.getByTestId("loader-spin");
    expect(spinner).toBeInTheDocument();
    expect(spinner.parentElement).toHaveStyle({
      marginTop: "10px",
      paddingTop: "20px",
      alignItems: "flex-start",
      width: "100%",
      display: "flex",
      justifyContent: "center",
    });
  });

  it("matches snapshot", () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });
});
