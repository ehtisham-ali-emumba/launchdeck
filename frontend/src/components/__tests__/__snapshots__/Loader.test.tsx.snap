// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader > matches snapshot 1`] = `
<div>
  <div
    class="sc-Qotzb kPZXnj"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning sc-fYsHOw fnXBHL css-dev-only-do-not-override-1m63z2v"
      data-testid="loader-spin"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;
